### ISSUE TITLE
USB soundcard not autodetected by PiDeck script

### ISSUE THREAD

**Issue by obbardc** [pideck/pideck#11]
USB soundcard comes up as `hw:1`, yet we load xwax with `hw:0` no matter what.

It would be nice to autodetect the first soundcard; and if no soundcards are found, show an error message rather than not showing xwax.

How about loading different settings depending on which soundcard is installed?

**Comment by obbardc** [comment]
No cards installed:

```
pi@pideck:~$ cat /proc/asound/cards 
--- no soundcards ---
pi@pideck:~$ ls -ld /proc/asound/card* | grep ^d
pi@pideck:~$ 
```

USB Card installed:

```
pi@pideck:~$ cat /proc/asound/cards
 1 [CODEC          ]: USB-Audio - USB AUDIO  CODEC
                      BurrBrown from Texas Instruments USB AUDIO  CODEC at usb-3f980000.usb-1.2, full
pi@pideck:~$ ls -ld /proc/asound/card* | grep ^d
dr-xr-xr-x 4 root root 0 Oct 15 09:18 /proc/asound/card1
```

AudioInjector card installed:

```
pi@pideck:~$ cat /proc/asound/cards
 0 [audioinjectorpi]: audioinjector-p - audioinjector-pi-soundcard
                      audioinjector-pi-soundcard
pi@pideck:~$ ls -ld /proc/asound/card* | grep ^d
dr-xr-xr-x 4 root root 0 Oct 15 09:21 /proc/asound/card0
```

Both cards installed:

```
pi@pideck:~$ cat /proc/asound/cards
 0 [audioinjectorpi]: audioinjector-p - audioinjector-pi-soundcard
                      audioinjector-pi-soundcard
 1 [CODEC          ]: USB-Audio - USB AUDIO  CODEC
                      BurrBrown from Texas Instruments USB AUDIO  CODEC at usb-3f980000.usb-1.2, full
pi@pideck:~$ ls -ld /proc/asound/card* | grep ^d
dr-xr-xr-x 4 root root 0 Oct 15 09:21 /proc/asound/card0
dr-xr-xr-x 4 root root 0 Oct 15 09:21 /proc/asound/card1
```

**Comment by obbardc** [comment]
See commit [ab9ca1ddcef8ae6ee2490da34caa8934905ed872](https://github.com/pideck/pideck/commit/ab9ca1ddcef8ae6ee2490da34caa8934905ed872) for a fix which works out of the box.

**Comment by obbardc** [comment]
> Depending on the USB device, the second line of output would probably not match 'BurrBrown, and may not even contain the letters 'USB'. So for a more generic test, we could check for 'USB-Audio' in the first line of output. Does the Audio Genie II device also have 'USB-Audio' on the first line of output? If not, perhaps we could check for 'usb-' on the second line.

The Audiogenie comes up as a "USB-Audio - USB AUDIO  CODEC" on the first line of the output. I wanted to allow separate parameters per USB card.

I agree it needs a bit of "massaging" to get right, including setting the fallback buffer size of 8ms, but it works.

Next step will probably be to move the card detection into a Python script rather than in the main script to tidy things up a bit.

**Comment by obbardc** [comment]
We can use `python-alsaaudio` debian package to do this.

**Comment by ximae** [comment]
Doesnt really need to be python, you could just make a separate bash script that gets called from the main startup script..... i was thinking in doing this already, and actually tidying up the autodetection part for when there is more supported cards. ( using a loop and an array to go through all the cards for the comparisons)

**Comment by danielhjames** [comment]
There are a relatively short list of interfaces which have phono pre-amps, we could aspire to supporting them all. Thanks to Paul who sent in this output for
![audio8dj](https://cloud.githubusercontent.com/assets/1867708/20708332/48bfc74c-b627-11e6-9fc3-063ea72ac421.jpg)
 the Audio 8 DJ.

**Comment by obbardc** [comment]
I think the easiest way will be making a python script to set each card up.

Cheers!

**Comment by ximae** [comment]
I have a traktor audio 6 which can also be supported. ill add the cat output sometime later.

also my working asound.conf ( i had to tweak the original for it work)

**Comment by ximae** [comment]
we really need them berhingers output though, those are likely the easiest cards to get a hold of.

**Comment by obbardc** [comment]
Yeah, setting volumes etc from a script is easy, but modifying system
files, like asound.conf, isn't something that will be as easy.

Cheers!

**Comment by ximae** [comment]
Just sent you guys an email with new scripts and included that audio 8 and my audio 6.
I have also forked the pideck repo if you prefer me to do a pull request.

**Comment by danielhjames** [comment]
Thanks! Please do make a pull request.

**Comment by beaucasque** [comment]
> See commit [ab9ca1ddcef8ae6ee2490da34caa8934905ed872](https://github.com/pideck/pideck/commit/ab9ca1ddcef8ae6ee2490da34caa8934905ed872) for a fix which works out of the box.

Hi! is it useful to add those script to a new pideck install and if so, how to?  Thanks!

**Comment by danielhjames** [comment]
Hi @beaucasque that is a really old commit, it was incorporated in early versions of PiDeck but is now superseded by the current git versions of these setup scripts.
